finality-aleph/src/compatibility.rs

```diff
@@ -0,0 +1,38 @@
+use codec::{Decode, Encode, Error, Input, Output};
+
+use crate::{network::Data, SessionId};
+
+#[derive(Encode, Eq, Decode, PartialEq, Debug, Copy, Clone)]
+pub struct Version(pub u16);
+
+pub trait Versioned {
+    const VERSION: Version;
+}
+
+/// Wrapper for data send over network. We need it to ensure compatibility.
+/// The order of the data and session_id is fixed in encode and the decode expects it to be data, session_id.
```

Comment:
Add "Since data is versioned, i.e. it's encoding starts with a version number in the standardized way, this will allow us to retrofit versioning here if we ever need to change this structure." or something like that.